openwisp_controller/config/admin.py

```diff
@@ -162,18 +162,24 @@ def _get_config_model(self):
 
     def _get_preview_instance(self, request):
         """
-        returns a temporary preview instance used for preview
+        returns a temporary instance used for preview
         """
         kwargs = {}
         config_model = self._get_config_model()
+        instance = config_model()
+        if request.POST.get('id'):
+            try:
+                instance = config_model.objects.get(pk=request.POST['id'])
+            except config_model.DoesNotExist:
+                pass
         for key, value in request.POST.items():
             # skip keys that are not model fields
             try:
                 field = config_model._meta.get_field(key)
             except FieldDoesNotExist:
                 continue
-            # skip m2m
-            if field.many_to_many:
+            # skip m2m or ID
+            if field.many_to_many or key == 'id':
```

Comment:
@nemesisdesign this was breaking resolution of variables in preview configuration. Skipping the ID field assign a random ID to the instance which makes all the related query return empty queryset. Removing this solves the issue for preview configuration view of both Device and VPN server.

Why was this needed?

Reply:
I wish I wrote a comment to remind myself but I didn't and now I am not sure :(